.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,41 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'r**'
+    types: [ labeled, synchronize ]
+
+jobs:
+  gpu-test:
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' || github.event_name == 'synchronize' }}
+
+    steps:
+    - name: Run nvidia-smi test
+      run: |
+        whoami
+        nvidia-smi
+
+    - name: Checkout the PR code
+      uses: actions/checkout@v3
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: Install NeMo-Curator and PyTest
+      run: |
+        pip install wheel cython
+        RAPIDS_NIGHTLY=1 pip install --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple ".[cuda12x]"
+        pip install pytest
+    
+    - name: Verify installation
+      run: |
+        pip list
+
+    - name: Run PyTests with GPU mark
+      run: |
+        python -m pytest -m gpu
```

Comment:
We run our GPU CI tests only on PyTests with the `pytest.mark.gpu` decorator. We already have quite a few of them in our repository:

- [exact_dedup_data](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_exact_dedup.py#L25)
- [TestSemDuplicates](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_semdedup.py#L51)
- [TestMetadataSep](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_seperate_by_metadata.py#L43)
- [TestMinhashes](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_fuzzy_dedup.py#L92)
- [TestLSH](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_fuzzy_dedup.py#L153)
- [TestFuzzyDuplicates](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_fuzzy_dedup.py#L210)
- [test_extract_partitioning_index](https://github.com/NVIDIA/NeMo-Curator/blob/main/tests/test_fuzzy_dedup.py#L463)